### PR TITLE
Add warning notice on GitHub Jobs Deprecation

### LINKED ISSUE #6574: Add a deprecation warning on the 'Available for hire' docs page
### What article on docs.github.com is affected?

https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/what-does-the-available-for-hire-checkbox-do

### What part(s) of the article would you like to see updated?

The page gives a great explanation of what the checkbox does, great! But what if the user doesn't head to jobs.github.com to see that the entire point of the checkbox and the docs' explanation didn't matter as It's shutting down? Now that's a problem.

It would be ideal for the page to be reflected with information stating it's currently deprecated, and again when its fully shut down.

### Additional information

EDIT: After further research, another two locations which may benefit from a similar change could be https://docs.github.com/en/github/getting-started-with-github/github-glossary#github-jobs and https://docs.github.com/en/github/finding-talent-with-github-jobs

### PR DESCRIPTION
**Pull request by rlaphoenix**
### Why:

Closes https://github.com/github/docs/issues/6574

### What's being changed:

It adds a warning box with information relating to the deprecation and eventual shutdown of jobs.github.com, linking to the blog post. All wording and typesetting are from the jobs.github.com deprecation warning banner.

### Check off the following:

- [X] I have reviewed my changes in staging (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**).
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by sarahs** [comment]
👋 We did some content restructuring that resulted in a conflict with your PR. It looks like we may not have access to check out your fork. Let us know if you need a hand resolving the conflict!

**Comment by rlaphoenix** [comment]
I've fixed the conflict, and I've also gone ahead and added the same warning message to https://docs.github.com/en/github/finding-talent-with-github-jobs/about-github-jobs

**Comment by janiceilene** [comment]
@rlaphoenix Thanks so much for opening a PR! I'll get this triaged for review ⚡

File: content/github/finding-talent-with-github-jobs/about-github-jobs.md [hunk 1]
```diff
@@ -10,6 +10,13 @@ topics:
   - Jobs
 ---
 
+{% warning %}
+
+GitHub Jobs is deprecated! New jobs will not be posted from May 19, 2021. It will shut down entirely on August 19, 2021.
```
Comment (jmarlena) [inline content/github/finding-talent-with-github-jobs/about-github-jobs.md:15]:
```suggestion
Deprecation note: GitHub Jobs is now deprecated. The last date to post a job was May 19, 2021. The GitHub Jobs site will shut down entirely on August 19, 2021. For more information, see the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```

File: content/github/finding-talent-with-github-jobs/about-github-jobs.md [hunk 2]
```diff
@@ -10,6 +10,13 @@ topics:
...
+Read more in the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```
Comment (jmarlena) [inline content/github/finding-talent-with-github-jobs/about-github-jobs.md:16]:
```suggestion

```

File: content/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/what-does-the-available-for-hire-checkbox-do.md [hunk 3]
```diff
@@ -9,6 +9,14 @@ versions:
 topics:
   - Accounts
 ---
+
+{% warning %}
+
+GitHub Jobs is deprecated! New jobs will not be posted from May 19, 2021. It will shut down entirely on August 19, 2021.
```
Comment (jmarlena) [inline content/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/what-does-the-available-for-hire-checkbox-do.md:15]:
```suggestion
Deprecation note: GitHub Jobs is now deprecated. The last date to post a job was May 19, 2021. The GitHub Jobs site will shut down entirely on August 19, 2021. For more information, see the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```

File: content/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/what-does-the-available-for-hire-checkbox-do.md [hunk 4]
```diff
@@ -9,6 +9,14 @@ versions:
...
+Read more in the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```
Comment (jmarlena) [inline content/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/what-does-the-available-for-hire-checkbox-do.md:16]:
```suggestion

```

**Review by jmarlena — APPROVED** [review]
Thanks for this thoughtful update!! I made some small style tweaks. 

This PR looks good to merge once tests pass. 🚢

**Pull request merged**